scripts/docker/hub/publish-docker.sh

```diff
@@ -0,0 +1,100 @@
+            --build-arg FROM_VERSION="${SCHEDULE_TAG}" \
+            --build-arg VCS_REF="${CI_COMMIT_SHA}" \
+            --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
+            --tag "parity/parity:${SCHEDULE_TAG}" \
+            --file tools/parity.Dockerfile .;
+        docker push "parity/parity:${SCHEDULE_TAG}";;
+    "beta")
+        echo "Docker TAGs - 'parity/ethereum:beta', 'parity/ethereum:latest', \
+            'parity/ethereum:${VERSION}-${CI_COMMIT_REF_NAME}'";
+        docker build --no-cache \
+            --build-arg VCS_REF="${CI_COMMIT_SHA}" \
+            --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
+            --tag "parity/ethereum:beta" \
+            --tag "parity/ethereum:latest" \
+            --tag "parity/ethereum:${VERSION}-${CI_COMMIT_REF_NAME}" \
+            --file tools/ethereum.Dockerfile .;
+        docker push "parity/ethereum:beta";
+        docker push "parity/ethereum:latest";
+        docker push "parity/ethereum:${VERSION}-${CI_COMMIT_REF_NAME}";
+
+        echo "Docker TAGs - 'parity/parity:beta', 'parity/parity:latest', \
+            'parity/parity:${VERSION}-${CI_COMMIT_REF_NAME}'";
+        docker build --no-cache \
+            --build-arg FROM_VERSION="${VERSION}-${CI_COMMIT_REF_NAME}" \
+            --build-arg VCS_REF="${CI_COMMIT_SHA}" \
+            --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
+            --tag "parity/parity:beta" \
+            --tag "parity/parity:latest" \
+            --tag "parity/parity:${VERSION}-${CI_COMMIT_REF_NAME}" \
+            --file tools/parity.Dockerfile .;
+        docker push "parity/parity:beta";
+        docker push "parity/parity:latest";
+        docker push "parity/parity:${VERSION}-${CI_COMMIT_REF_NAME}";;
+    "stable")
+        echo "Docker TAGs - 'parity/ethereum:${VERSION}-${CI_COMMIT_REF_NAME}', 'parity/ethereum:stable'";
+        docker build --no-cache \
+            --build-arg VCS_REF="${CI_COMMIT_SHA}" \
+            --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
+            --tag "parity/ethereum:${VERSION}-${CI_COMMIT_REF_NAME}" \
+            --tag "parity/ethereum:stable" \
+            --file tools/ethereum.Dockerfile .;
+        docker push "parity/ethereum:${VERSION}-${CI_COMMIT_REF_NAME}";
+        docker push "parity/ethereum:stable";
+
+        echo "Docker TAGs - 'parity/parity:${VERSION}-${CI_COMMIT_REF_NAME}', 'parity/parity:stable'";
+        docker build --no-cache \
+            --build-arg FROM_VERSION="${VERSION}-${CI_COMMIT_REF_NAME}" \
+            --build-arg VCS_REF="${CI_COMMIT_SHA}" \
+            --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
+            --tag "parity/parity:${VERSION}-${CI_COMMIT_REF_NAME}" \
+            --tag "parity/parity:stable" \
+            --file tools/parity.Dockerfile .;
+        docker push "parity/parity:${VERSION}-${CI_COMMIT_REF_NAME}";
+        docker push "parity/parity:stable";;
+    *)
```

Comment:
this would be the tag released versions then, right? don't you want to use the version and only the version for the tag?

Reply:
tags and PR numbers will go there. So it will be like `parity/ethereum:2.5.0-beta` or `parity/ethereum:2.5.0-pr-10644`. Least is actually not desired to end up in registry, but for the testing purposes it'll be there for a while. That's why I covered all the cases.